python/lsst/ci/build.py

```diff
@@ -328,15 +332,59 @@ def write_status(self):
             yaml.dump(status, sf, encoding="utf-8", default_flow_style=False)
 
     @staticmethod
-    def run(args):
+    def get_agent() -> str:
+        """
+        Determine the system agent based on OS type and architecture.
+
+        Returns
+        -------
+        str
+            The formatted agent string, such as "apple_arm64" or
+            "linux_x86_64". If the system platform or architecture
+            cannot be determined, returns "unknown_agent".
+        """
+        os_type = sys.platform
+        arch = platform.machine()
+
+        if not os_type or not arch:
+            agent = "unknown_agent"
+        else:
+            match os_type:
+                case "darwin":
+                    agent = f"apple_{arch}"
+                case _:
+                    agent = f"{os_type}_{arch}"
+
+        # Print verification message
+        print("Agent is unknown." if agent == "unknown_agent" else f"Agent verified: {agent}")
+
+        return agent
+
+    @staticmethod
+    def run(args) -> None:
+        """Run the build process, including GitHub status updates."""
+        agent = Builder.get_agent()
+
         # Ensure build directory exists and is writable
         build_dir = args.build_dir
         if not os.access(build_dir, os.W_OK):
             raise Exception(f"Directory {build_dir!r} does not exist or isn't writable.")
 
+        # Load PR pydantic model created in prepare.py
+        manager = PRInfoManager(build_dir)
+        pr_info = manager.load_pr_info()
+        # pr_info = Builder.load_pr_info(build_dir)
+
+        if not pr_info.items:
+            print("WARNING: PR information could not be loaded. Skipping posting Jenkins statuses on GitHub.")
+        else:
+            for info in pr_info:
```

Comment:
Does this work? Shouldn't it be `pr_info.items` that you are looping over? Otherwise it will show the contents of the model as a single tuple. Would be more intuitive to add `__iter__` method to the model to make it work how you expect. Why isn't this using the `summary` method though?

Also consider adding `__str__` to the individual PR models and using that in the `summary` method.